File: Difficulty: Basic/Remove common characters and concatenate/remove-common-characters-and-concatenate.py
# User function Template for python3

class Solution:
    
    #Function to remove common characters and concatenate two strings.
    def concatenatedString(self, s1, s2):
        result = ""
        
        for ch in s1:
            if ch not in s2:
                result += ch
        
        for ch in s2:
            if ch not in s1:
                result += ch
        
        if result == "":
            return "-1"
        else:
            return result

File: Difficulty: Basic/Remove common characters and concatenate/remove-common-characters-and-concatenate.cpp
class Solution {
  public:
    string concatenatedString(string s1, string s2) {
        unordered_set<char> set1(s1.begin(), s1.end());
        unordered_set<char> set2(s2.begin(), s2.end());
        
        string result = "";

        for (char c : s1) {
            if (set2.find(c) == set2.end()) {
                result.push_back(c);
            }
        }
        
        for (char c : s2) {
            if (set1.find(c) == set1.end()) {
                result.push_back(c);
            }
        }
        
        if (result.empty()) {
            return "-1";
        } else {
            return result;
        }
    }
};